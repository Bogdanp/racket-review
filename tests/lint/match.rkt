#lang racket/base

(require racket/match)

(define foo 1)

(match foo
  [1
   (define foo 2)
   (define bar 3)
   (+ foo bar)]

  [2
   (define bar 4)
   bar])

(match foo
  [1
   (define bar 3)
   bar])

(match foo
  [null 1]
  [_ 2])

(match foo
  [(? eof-object?) void]
  [(or (? eof-object?) foo) (void)])

(match (list 1 2)
  [`(1 ,a) void]
  [`,b void])
