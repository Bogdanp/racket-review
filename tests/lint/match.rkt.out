/match.rkt:9:12:warning:identifier 'foo' shadows an earlier binding
/match.rkt:23:4:error:use '() for match pattern instead of null or empty
/match.rkt:28:24:warning:identifier 'foo' is never used
/match.rkt:31:9:warning:identifier 'a' is never used
/match.rkt:32:6:warning:identifier 'b' is never used
